pkg/ocgormv2/context.go

```diff
@@ -0,0 +1,18 @@
+package ocgormv2
+
+import (
+	"context"
+
+	"gorm.io/gorm"
+)
+
+var (
+	contextScopeKey = "_opencensusContext"
+)
+
+// WithContext sets the current context in the db instance for instrumentation.
+func WithContext(ctx context.Context, db *gorm.DB) *gorm.DB {
+	// we MUST call `Session` here because otherwise resulting *gorm.DB will be unsafe
+	// to reuse, see https://gorm.io/docs/method_chaining.html for more information
+	return db.Set(contextScopeKey, ctx).Session(&gorm.Session{})
```

Comment:
Doesn't this move all queries into a transaction? Just reading through the [.Session](https://github.com/go-gorm/gorm/blob/v1.25.4/gorm.go#L218) method.

Reply:
Ahh WithContext already does that, so this should represent no change for consumers. I agree with this patch.